sound/soc/sof/sof-audio.h

```diff
@@ -40,6 +40,11 @@ struct snd_sof_widget;
 struct snd_sof_route;
 struct snd_sof_control;
 
+struct snd_sof_dai_config_data {
+	int dai_index;
+	int id;
```

Comment:
You know me @ranj063. 
I will always ask the same question: what is the difference between index and id?
Someone will invert the two.
Please consider renaming these fields and remove any sort of ambiguity.

I also see that even 'id' is used for very different things:

config->hda.link_dma_ch = data->id;
config->alh.stream_id = data->id;

And last but not least, in ASoC the configurations are at the dai_link level, but in the SOF firmware they relate to a dai, so this could get confusing even faster.

Reply:
@plbossart the ID is left ambiguous because different DAI types could use the ID field differently. In the case of the ALH DAI's, it represents stream ID and in the case of HDA it represents the channel ID. If we want to remove the ambiguity, I'd have to intriduce 2 separate fields and one goes unused for each of the 2 types. What do you prefer?

Reply:
call it dai_data then, and add a comment that this field might contain information that's specific to the type of DAI.

Reply:
done!